cmd/openshift-install/create.go

```diff
@@ -289,49 +289,36 @@ func destroyBootstrap(ctx context.Context, config *rest.Config, directory string
 	events := client.CoreV1().Events("kube-system")
 
 	eventTimeout := 30 * time.Minute
-	logrus.Infof("Waiting up to %v for the bootstrap-complete event...", eventTimeout)
-	eventContext, cancel := context.WithTimeout(ctx, eventTimeout)
-	defer cancel()
-	_, err = Until(
-		eventContext,
-		"",
-		func(sinceResourceVersion string) (watch.Interface, error) {
-			for {
-				watcher, err := events.Watch(metav1.ListOptions{
-					ResourceVersion: sinceResourceVersion,
-				})
-				if err == nil {
-					return watcher, nil
-				}
-				select {
-				case <-eventContext.Done():
-					return watcher, err
-				default:
-					logrus.Warningf("Failed to connect events watcher: %s", err)
-					time.Sleep(2 * time.Second)
-				}
-			}
-		},
-		func(watchEvent watch.Event) (bool, error) {
-			event, ok := watchEvent.Object.(*corev1.Event)
-			if !ok {
-				return false, nil
-			}
+	logrus.Infof("Waiting up to %v for the bootstrap-success event...", eventTimeout)
+	if err := wait.PollImmediate(time.Second, eventTimeout, func() (done bool, err error) {
+		if _, err = events.Get("bootstrap-success", metav1.GetOptions{}); err == nil {
+			return true, nil
+		}
+		logrus.Debugf("Failed to get bootstrap-success event: %v", err)
+		return false, nil
+	}); err != nil {
+		return errors.Wrap(err, "waiting for bootstrap-success")
+	}
 
-			if watchEvent.Type == watch.Error {
-				logrus.Debugf("error %s: %s", event.Name, event.Message)
-				return false, nil
-			}
+	logrus.Info("Taking bootstrap node out of service...")
+	if err := destroybootstrap.Disable(rootOpts.dir); err != nil {
+		return errors.Wrap(err, "taking bootstrap out of service")
+	}
 
-			if watchEvent.Type != watch.Added {
-				return false, nil
-			}
+	// TODO: poll ELB that the bootstrap node really updated
 
-			logrus.Debugf("added %s: %s", event.Name, event.Message)
-			return event.Name == "bootstrap-complete", nil
-		},
-	)
-	if err != nil {
+	if _, err := events.Create(makeBootstrapOutOfServiceEvent("kube-system", "bootstrap-outofservice")); err != nil && !apierrors.IsAlreadyExists(err) {
```

Comment:
Based on @abhinavdahiya's comment about not wanting to be a writer, I think that if we made this optional (don't fail on an error sending), then we could satisfy both cases.  In the common case, we would be faster and avoid confusing clients.  In the case where the installer cannot write events, we'll be slower, but the installer can still proceed.

Reply:
+1